content/references/rippled-api/api-conventions/serialization.md

```diff
@@ -0,0 +1,285 @@
+
+### Definitions File
+
+The following JSON file defines the important constants you need for serializing XRP Ledger data to its binary format and deserializing it from binary:
+
+**<https://github.com/ripple/ripple-binary-codec/blob/master/src/enums/definitions.json>**
+
+The following table defines the top-level fields from the definitions file:
+
+| Field                 | Contents                                             |
+|:----------------------|:-----------------------------------------------------|
+| `TYPES`               | Map of data types to their "type code" for constructing field IDs and sorting fields in canonical order. Codes below 1 should not appear in actual data; codes above 10000 represent special "high-level" object types such as "Transaction" that cannot be serialized inside other objects. |
+| `LEDGER_ENTRY_TYPES`  | Map of [ledger objects](ledger-object-types.html) to their data type. These appear in ledger state data, and in the "affected nodes" section of processed transactions' [metadata](transaction-metadata.html). |
+| `FIELDS`              | A sorted array of tuples representing all fields that may appear in transactions, ledger objects, or other data. The first member of each tuple is the string name of the field and the second member is an object with that field's properties. (See the "Field properties" table below for definitions of those fields.) |
+| `TRANSACTION_RESULTS` | Map of [transaction result codes](transaction-results.html) to their numeric values. Result types not included in ledgers have negative values;`tesSUCCESS` has numeric value 0; [`tec`-class codes](tec-codes.html) represent failures that are included in ledgers. |
+| `TRANSACTION_TYPES`   | Map of all [transaction types](transaction-types.html) to their numeric values. |
+
+For purposes of serializing transactions for signing and submitting, the `FIELDS`, `TYPES`, and `TRANSACTION_TYPES` fields are necessary.
+
+The field definition objects in the `FIELDS` array have the following fields:
+
+| Field            | Type    | Contents                                        |
+|:-----------------|:--------|:------------------------------------------------|
+| `nth`            | Number  | The [field code](#field-codes) of this field, for use in constructing its [Field ID](#field-ids) and ordering it with other fields of the same data type. |
+| `isVLEncoded`    | Boolean | If `true`, this field is [variable-length encoded](#variable-length-encoding). |
+| `isSerialized`   | Boolean | If `true`, this field should be encoded into serialized binary data. When this field is `false`, the field is typically reconstructed on demand rather than stored. |
+| `isSigningField` | Boolean | If `true` this field should be serialized when preparing a transaction for signing. If `false`, this field should be omitted from the data to be signed. (It may not be part of transactions at all.) |
```

Comment:
If this is what we mean, perhaps consider changing:

> It may not be part of transactions at all.

to

> It must not be part of transactions at all.

Reply:
Not what we mean.

Some "non-signing" fields are not part of transactions at all. But some are parts of transactions that merely aren't signed. (For example, you can't sign the signature for chicken-egg reasons.)

Reply:
Thank you for taking the time to explain, Rome - got it! 🙌